src/main/java/com/unstoppabledomains/resolution/ENS.java

```diff
@@ -32,42 +34,52 @@ Boolean isSupported(String domain) {
 
   @Override
   String addr(String domain, String ticker) throws NamingServiceException {
-    // TODO Auto-generated method stub
-    return null;
-  }
-
-  @Override
-  String ipfsHash(String domain) throws NamingServiceException {
-    String resolverAddress = getResolverAddress(domain);
-    Resolver resolverContract = (Resolver)buildContract(resolverAddress, EnsContractType.Resolver); 
-    return resolverContract.getRecord(domain, "gundb.username.value");
+    if (!ticker.equalsIgnoreCase("ETH")) {
+      throw new NamingServiceException(NSExceptionCode.UnsupportedCurrency, new NSExceptionParams("c", ticker.toUpperCase()));
+    }
+    Resolver resolver = getResolverContract(domain);
+    byte[] tokenId = this.tokenId(domain);
```

Comment:
`this` is redundant here (generally, a constructor is the most frequent place for usage of `this` keyword).